CMakeLists.txt

```diff
@@ -479,6 +479,17 @@ if(LINUX)
 endif()
 
 if(MSVC)
+  # MSVC by default does not apply the correct __cplusplus version as specified by the C++ standard
```

Comment:
Resolving Windows build issue, see https://github.com/NVIDIA/cutlass/issues/1171#issuecomment-1792411693

Reply:
If this does fix the problem I am not sure if we want to set this global, @malfet will likely have thoughts on this.

That being said it is possible to set flags on specific targets, see:
https://github.com/pytorch/pytorch/blob/d084a024aede3b2420c5977dd473988c63e9b9f6/caffe2/CMakeLists.txt#L1820-L1832

Reply:
As mentioned in the chat, I think we want to set this globally, since it restores MSVC to C++-Standard compliant behavior. 

See 
 * [C++ Standard](https://en.cppreference.com/w/cpp/preprocessor/replace#Predefined_macros)
 * [MSVC Docs](https://learn.microsoft.com/en-us/cpp/build/reference/zc-cplusplus?view=msvc-170) which pretty clearly states that the default behavior of MSVC is a hack.